src/components/DataTable.js

```diff
@@ -73,7 +75,23 @@ export default ajaxCaller(class DataTable extends Component {
 
     return h(Fragment, [
       !!entities && h(Fragment, [
-        childrenBefore && childrenBefore({ entities, columnSettings }),
+        div({ style: { display: 'flex' } }, [
+          childrenBefore && childrenBefore({ entities, columnSettings }),
+          div({ style: { flexGrow: 1 } }),
+          div({ style: { display: 'flex' } }, [
+            h(SearchInput, {
+              style: { width: 300, borderColor: colors.gray[3], borderRadius: '4px 0 0 4px' },
+              placeholder: 'Filter',
+              onChange: ({ target: { value } }) => this.setState({ textFilter: value }),
+              onSearch: () => this.loadData(),
+              value: textFilter
+            }),
+            buttonPrimary({
```

Comment:
TOL: It seems like we're actually introducing two new components in this change, 1. a text box that happens to be `type=search`, and 2. a special search input with lazy callback behavior and an explicit search affordance. All the complexity is around that second case, which is (currently) only used here. I wonder if we should make that an explicit distinction, so that the other inputs don't need to pull in all the underlying complexity.

Just a thought, not necessarily something to do right now.